src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +240,129 @@ pub fn register(registry: &mut FunctionRegistry) {
+                    return None;
+                }
+            }
+            DataType::EmptyMap => (DataType::EmptyMap, DataType::EmptyMap),
+            _ => {
+                return None;
+            }
+        };
+
+        if map_key_type != DataType::EmptyMap {
+            for arg_type in args_type.iter().skip(1) {
+                if arg_type != &map_key_type {
+                    return None;
+                }
+            }
+        } else {
+            let key_type = &args_type[1];
+            if !key_type.is_boolean()
+                && !key_type.is_string()
+                && !key_type.is_numeric()
+                && !key_type.is_decimal()
+                && !key_type.is_date_or_date_time()
+            {
+                return None;
+            }
+            for arg_type in args_type.iter().skip(2) {
+                if arg_type != key_type {
+                    return None;
+                }
+            }
+        }
+
+        let return_type = args_type[0].clone();
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_delete".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: return_type.clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(|_, args_domain| {
+                    FunctionDomain::Domain(args_domain[0].clone())
+                }),
+                eval: Box::new(move |args, _ctx| {
+                    let input_length = args.iter().find_map(|arg| match arg {
+                        ValueRef::Column(col) => Some(col.len()),
+                        _ => None,
+                    });
+
+                    let mut output_map_builder =
+                        ColumnBuilder::with_capacity(&return_type, input_length.unwrap_or(1));
+
+                    for idx in 0..(input_length.unwrap_or(1)) {
+                        let mut delete_key_list = HashSet::new();
+
+                        for input_key_item in args.iter().skip(1) {
+                            let input_key = match &input_key_item {
+                                ValueRef::Scalar(scalar) => scalar.clone(),
+                                ValueRef::Column(col) => unsafe { col.index_unchecked(idx) },
+                            };
+
+                            delete_key_list.insert(input_key.to_owned());
+                        }
+
+                        let input_map_sref = match &args[0] {
+                            ValueRef::Scalar(map) => map.clone(),
+                            ValueRef::Column(map) => unsafe { map.index_unchecked(idx) },
+                        };
+
+                        let inner_builder_type = match input_map_sref.infer_data_type() {
+                            DataType::Map(box typ) => typ,
+                            DataType::EmptyMap => DataType::EmptyMap,
+                            _ => unreachable!(),
+                        };
```

Comment:
those codes can be moved to line 334, as if the value is `Null` or `EmptyMap`, we don't need build `delete_key_list`.